sdk/src/config/context_config.rs

```diff
@@ -0,0 +1,208 @@
+use super::*;
+use alloc::string::String;
+use alloy_primitives::Address;
+use core::fmt::Debug;
+use core::net::IpAddr;
+use gadget_io::SupportedChains;
+use libp2p::Multiaddr;
+use serde::{Deserialize, Serialize};
+use url::Url;
+
+#[derive(Debug, Default, Clone, clap::Parser, Serialize, Deserialize)]
+#[command(name = "General CLI Context")]
+#[cfg(feature = "std")]
+pub struct ContextConfig {
+    /// Pass through arguments to another command
+    #[command(subcommand)]
+    pub gadget_core_settings: GadgetCLICoreSettings,
+}
+
+#[derive(Debug, Clone, clap::Parser, Serialize, Deserialize)]
+#[cfg(feature = "std")]
+pub enum GadgetCLICoreSettings {
+    #[command(name = "run")]
+    Run {
+        #[arg(long, short = 'b', env)]
+        bind_addr: IpAddr,
+        #[arg(long, short = 'p', env)]
+        bind_port: u16,
+        #[arg(long, short = 't', env)]
+        test_mode: bool,
+        #[arg(long, short = 'l', env)]
+        log_id: Option<String>,
+        #[arg(long, env)]
+        #[serde(default = "gadget_io::defaults::http_rpc_url")]
+        http_rpc_url: Url,
+        #[arg(long, env)]
+        #[serde(default = "gadget_io::defaults::ws_rpc_url")]
+        ws_rpc_url: Url,
+        #[arg(long, value_parser = <Multiaddr as std::str::FromStr>::from_str, action = clap::ArgAction::Append, env)]
+        #[serde(default)]
+        bootnodes: Option<Vec<Multiaddr>>,
+        #[arg(long, short = 'd', env)]
+        keystore_uri: String,
+        #[arg(long, value_enum, env)]
+        chain: SupportedChains,
+        #[arg(long, short = 'v', global = true, action = clap::ArgAction::Count)]
+        verbose: u8,
+        /// Whether to use pretty logging
+        #[arg(long, env)]
+        pretty: bool,
+        #[arg(long, env)]
+        keystore_password: Option<String>,
+        /// The protocol to use
+        #[arg(long, value_enum, env)]
+        #[serde(default = "default_protocol")]
+        protocol: Protocol,
+        /// The blueprint ID for Tangle protocol
```

Comment:
Just a note, this can probably be improved with arg groups. Here's an example:
https://github.com/tangle-network/gadget/blob/2111dc9bb042f0c3e2e9d8f3d2db994fa8631d56/cli/src/create.rs#L4-L23
Where `--repo` (and its additional options) cannot be used alongside `--path`.